packages/beacon-node/src/chain/seenCache/seenGossipBlockInput.ts

```diff
@@ -123,28 +138,34 @@ export class SeenGossipBlockInput {
         resolveAvailability(allBlobs);
         metrics?.syncUnknownBlock.resolveAvailabilitySource.inc({source: BlockInputAvailabilitySource.GOSSIP});
         const {blobs, blobsBytes} = allBlobs;
+        const blockInput = getBlockInput.postDeneb(
+          config,
+          signedBlock,
+          BlockSource.gossip,
+          blobs,
+          blockBytes ?? null,
+          blobsBytes
+        );
+
+        resolveBlock(blockInput);
         return {
-          blockInput: getBlockInput.postDeneb(
-            config,
-            signedBlock,
-            BlockSource.gossip,
-            blobs,
-            blockBytes ?? null,
-            blobsBytes
-          ),
+          blockInput,
           blockInputMeta: {pending: null, haveBlobs: blobs.length, expectedBlobs: blobKzgCommitments.length},
         };
       } else {
+        const blockInput = getBlockInput.blobsPromise(
+          config,
+          signedBlock,
+          BlockSource.gossip,
+          blobsCache,
+          blockBytes ?? null,
+          availabilityPromise,
+          resolveAvailability
+        );
+
+        resolveBlock(blockInput);
```

Comment:
in this case there are not enough blobs showing up, why do we `resolveBlock()` here? if it's needed in both `if` and `else` then let's put in common place

Reply:
so that the waiting handler for blob can resolve and emit event to pull blobs (and add its peer over there)

For code cleanliness its ok to be replicated as the return follows this statement